all-actors/src/main/java/org/sunbird/BaseActor.java

```diff
@@ -0,0 +1,106 @@
+package org.sunbird;
+
+import akka.actor.UntypedAbstractActor;
+import org.apache.log4j.Logger;
+import org.sunbird.message.IResponseMessage;
+import org.sunbird.message.Localizer;
+import org.sunbird.message.ResponseCode;
+import org.sunbird.request.Request;
+
+import java.util.Locale;
+
+/**
+ * @author Amit Kumar
+ */
+public abstract class BaseActor extends UntypedAbstractActor {
+
+    private Logger logger = Logger.getLogger(BaseActor.class);
+    public abstract void onReceive(Request request) throws Throwable;
+    protected Localizer localizer = Localizer.getInstance();
+
+    @Override
+    public void onReceive(Object message) throws Throwable {
+        if (message instanceof Request) {
+            Request request = (Request) message;
+            String operation = request.getOperation();
+            logger.info("BaseActor:onReceive called for operation:" + operation);
+            try {
+                logger.info(String.format("%s:%s:method started at %s",this.getClass().getSimpleName(),operation,System.currentTimeMillis()));
```

Comment:
String.format is used when we have to pass dynamic value. applicable for all places.